packages/knip/src/util/modules.ts

```diff
@@ -15,7 +15,7 @@ export const getPackageNameFromFilePath = (value: string) => {
   return value;
 };
 
-export const isStartsLikePackageName = (specifier: string) => /^@?[a-z0-9]/.test(specifier);
+export const isStartsLikePackageName = (specifier: string) => /^@?[a-z0-9._]/.test(specifier);
```

Comment:
```suggestion
export const isStartsLikePackageName = (specifier: string) => /^(@[a-z0-9._]|[a-z0-9])/.test(specifier);
```

Regex could be improved perhaps, but at least we're not allowing non-scoped packages to start with `.` or `_`.

Reply:
Thanks to chatGPT, we could improve the regex like this

```suggestion
export const isStartsLikePackageName = (specifier: string) => /^(?:@([._]?[a-z0-9-]{1,213})\/)?[a-z0-9]([a-z0-9-_.]*)(\/[a-z0-9]([a-z0-9-_.]*))*$/gm.test(specifier);
```

Regex tests here https://regex101.com/r/Ukm3Fw/2

Or you could use this package https://www.npmjs.com/package/validate-npm-package-name

Reply:
It's ok for me use your suggested version